### PR TITLE
 [iOS11] Fix warnings

### PR DESCRIPTION
**Pull request by Eke**
Hello!

I fixed some warnings in Xcode 9 and iOS 11. Also re-enabled iOS and tvOS support for `PINCache` in `PINRemoteImage.podspec`, i think it was disabled in #421 .

After that PR was merged users got error(was reported in Texture slack channel too) :
```
[!] The platform of the target `App name` (iOS 8.0) is not compatible with `PINRemoteImage/PINCache (3.0.0-beta.13)`, which does not support `ios`.
```

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: PINRemoteImage.podspec [hunk 1]
```diff
@@ -76,7 +76,9 @@ Pod::Spec.new do |s|
   s.subspec "PINCache" do |pc|
     pc.dependency 'PINRemoteImage/Core'    
   	pc.dependency 'PINCache', '=3.0.1-beta.6'
-	pc.osx.deployment_target = osx_deployment
+    pc.ios.deployment_target = ios_deployment
```
Comment (garrettmoon) [inline PINRemoteImage.podspec]: Super minor nit: mind fixing the spacing?

Comment (Eke) [reply PINRemoteImage.podspec]: no problem, will fix tomorrow morning. Thank you for review

File: Source/Classes/PINURLSessionManager.m [hunk 2]
```diff
@@ -209,9 +209,18 @@ - (BOOL)responseRecoverableFrom404:(NSHTTPURLResponse*)response
 #if DEBUG
 - (void)concurrentDownloads:(void (^_Nullable)(NSUInteger concurrentDownloads))concurrentDownloadsCompletion
 {
-    [self.session getAllTasksWithCompletionHandler:^(NSArray<__kindof NSURLSessionTask *> * _Nonnull tasks) {
-        concurrentDownloadsCompletion(tasks.count);
-    }];
+    if (@available(macos 10.11, iOS 9.0, watchOS 2.0, tvOS 9.0,  *)) {
```
Comment (maicki) [inline Source/Classes/PINURLSessionManager.m]: @garrettmoon I remember we talked about not using `@available` to support Xcode < 9. Is this still the case?

Comment (garrettmoon) [reply Source/Classes/PINURLSessionManager.m]: @maicki yeah, still shooting for that, but it'll be on top of this and will be a macro which uses `@availabile` on Xcode 9.

**Pull request merged**